src/model/model_json.nit

```diff
@@ -0,0 +1,311 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Make model entities Jsonable.
+#
+# To avoid cycles, every reference from a MEntity to another is replaced by a
+# MEntityRef.
+#
+# How subobjects are retrieved using the MEntityRef is the responsability of the
+# client. Json objects can be returned as this or inflated with concrete objet
+# rather than the refs.
+#
+# TODO consider serialization module?
+module model_json
+
+import model::model_collect
+import json
+import loader
+
+# A reference to another mentity.
+class MEntityRef
+	super Jsonable
+
+	# MEntity to link to.
+	var mentity: MEntity
+
+	# Reference are not inflated by default.
+	var inflated = false is writable
+
+	# Return `self` as a Json Object.
+	fun json: JsonObject do
+		var obj = new JsonObject
+		obj["full_name"] = mentity.full_name
+		return obj
+	end
+
+	redef fun to_json do
+		if inflated then
+			return mentity.to_json
+		end
+		return json.to_json
+	end
+end
+
+redef class MEntity
+	super Jsonable
+
+	# Create a ref to a `mentity`.
+	fun to_ref(mentity: nullable MEntity): nullable MEntityRef do
+		if mentity == null then return null
+		return new MEntityRef(mentity)
+	end
+
+	# Return a collection of `mentities` as a JsonArray of MEntityRefs.
+	fun to_refs(mentities: Collection[MEntity]): JsonArray do
```

Comment:
None, but since I cannot define the méthode on Collection[MEntity] I had to define it somewhere.

Better idea?